src/main/kotlin/sg/com/quantai/etl/controllers/CryptoController.kt

```diff
@@ -0,0 +1,48 @@
+package sg.com.quantai.etl.controllers
+
+import com.fasterxml.jackson.databind.JsonNode
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import sg.com.quantai.etl.services.CryptoService
+
+@RestController
+@RequestMapping("/crypto")
+class CryptoController(private val cryptoService: CryptoService) {
+
+    /**
+     * Fetch the current price of a cryptocurrency
+     * Example Usage:
+     * GET /crypto/current/price?symbol=BTC&currency=USD
+     */
+    @GetMapping("/price/current")
+    fun getCryptoPrice(
+        @RequestParam symbol: String,
+        @RequestParam currency: String
+    ): ResponseEntity<JsonNode> {
+        val response = cryptoService.fetchCryptoPrice(symbol, currency)
+        return if (response != null) {
+            ResponseEntity.ok(response)
+        } else {
+            ResponseEntity.status(500).body(null)
+        }
+    }
+
+    /**
+     * Fetch and store historical cryptocurrency data
+     * Example Usage:
+     * POST /crypto/historical/store?symbol=BTC&currency=USD&limit=10
+     */
+    @PostMapping("/historical/store")
```

Comment:
To follow the pattern created by Karthik during the newsArticle, we should:
- Call the endpoint by "/source/action". Therefore, instead of having it as "/historical/store", we will have it as "/cryptocurrency/fetch"

Reply:
Done 
Renamed to /cryptocurrency/fetch